GUI/MainWindow.xaml.cs

```diff
@@ -351,8 +359,21 @@ private void RunAllTasks(string v)
         {
             dynamicTasksObservableCollection = new ObservableCollection<InRunTask>();
 
+
             for (int i = 0; i < staticTasksObservableCollection.Count; i++)
-                dynamicTasksObservableCollection.Add(new InRunTask("Task" + (i + 1) + staticTasksObservableCollection[i].metaMorpheusTask.TaskType, staticTasksObservableCollection[i].metaMorpheusTask));
+            {
+                dynamicTasksObservableCollection.Add(new InRunTask("Task" + (i + 1) + "-" + staticTasksObservableCollection[i].metaMorpheusTask.TaskType, staticTasksObservableCollection[i].metaMorpheusTask));
+                if (staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID != null 
+                    && staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID != "SearchTask"
+                    && staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID != "GPTMDTask"
+                    && staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID != "CalibrateTask"
+                    && staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID != "XLSearchTask")
+                {
+                    dynamicTasksObservableCollection[i].DisplayName = "Task" + (i + 1) + "-" + staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID;
+                    dynamicTasksObservableCollection[i].Id = "Task" + (i + 1) + "-" + staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID;
```

Comment:
dynamicTasksObservableCollection[i].Id = staticTasksObservableCollection[i].metaMorpheusTask.CommonParameters.TaskID

Reply:
Shouldn't we have this?